### ISSUE TITLE
iOS 15.4 Webgl via Metal issue

### ISSUE THREAD

**Issue by jeroenvheel** [openlayers/openlayers#13486]
**Describe the bug**
The Webgl layer is not panning / zooming with the map.
In iOS 15.4 the Webgl via Metal is enabled by default.

**To Reproduce**
1. Go to https://viewer.bomenwacht.nl/?code=935771DB2CB1AF76AD30B1B4345405E2 with a iOS device.
2. Try to pan / zoom te map

**Expected behavior**
When you turn off Webgl via Metal in the experimental features the Webgl layer is panning / zooming again with the map.

**Comment by ahocevar** [comment]
The official OpenLayers examples work fine on iOS 15.4. Maybe you are using an older version of OpenLayers, or there is something in your code which causes that behavior.

**Comment by v4lli** [comment]
@ahocevar I can confirm that, since iOS 15.4, some WebGL TileLayers are broken. The official [WebGL Sea Level](https://openlayers.org/en/latest/examples/webgl-sea-level.html) example is one example of this:

![IMG_9686B8D68BC2-1](https://user-images.githubusercontent.com/1577223/158721716-f32f396e-25fc-4dc8-a677-570c90131e82.jpeg)

As noted by @jeroenvheel, disabling "Webgl via Metal" in the Safari advanced settings fixes this (I think it has become the default since iOS 15.4).

**Comment by jeroenvheel** [comment]
@ahocevar @v4lli also the Layer swipe example is broken on iOS 15.4

https://openlayers.org/en/latest/examples/webgl-layer-swipe.html

**Comment by jeroenvheel** [comment]
I did not found any error in the console wile debugging my iphone / ipad

the way i create the Webgl layer is like this.

```ts
    private _createInspectionLayers(): void {
        this._inspectionSource = new VectorSource();
        this._inspectionLayer = new WebGLPointsLayer({
            source: this._inspectionSource,
            style: {
                symbol: {
                    symbolType: 'image',
                    src: this._layerStyleService.toInspecStyle,
                    size: ['interpolate', ['exponential', 2.4], ['zoom'], 2, 1, 30, 55],
                    rotateWithView: false,
                    offset: [0, 0]
                }
            },
            disableHitDetection: false,
            zIndex: 5
        });
        this._inspectedSource = new VectorSource();
        this._inspectedLayer = new WebGLPointsLayer({
            source: this._inspectedSource,
            style: {
                symbol: {
                    symbolType: 'image',
                    src: this._layerStyleService.inspectedStyle,
                    size: ['interpolate', ['exponential', 2.5], ['zoom'], 2, 1, 23, 45],
                    rotateWithView: false,
                    offset: [0, 0]
                }
            },
            disableHitDetection: false,
            zIndex: 4
        });
    }
```
for the src property i create a dataUrl and cache it in the property inspectedStyle
```ts
  protected _circleStyle(outerColor: string, innerColor: string, innerRadius = 14): string {
    const canvas = document.createElement('canvas');
    canvas.setAttribute('height', '40px');
    canvas.setAttribute('width', '40px');
    const ctx = <CanvasRenderingContext2D>canvas.getContext('2d');

    // start outer
    ctx.beginPath();
    ctx.arc(20, 20, 18, 0, 2 * Math.PI);
    ctx.strokeStyle = outerColor;
    ctx.fillStyle = outerColor;
    ctx.stroke();
    ctx.fill();
    ctx.closePath();

    // start inner
    ctx.beginPath();
    ctx.arc(20, 20, innerRadius, 0, 2 * Math.PI);
    ctx.strokeStyle = innerColor;
    ctx.fillStyle = innerColor;
    ctx.stroke();
    ctx.fill();
    ctx.closePath();

    return canvas.toDataURL();
  }
```

**Comment by ahocevar** [comment]
Thanks for reporting all findings so far. Looks like it only happens when style expressions are used. Will be hard to investigate this further. Maybe it's something in OpenLayers, maybe it's a Safari problem. I can reproduce it on iOS, but not on MacOS.

Any help with this, any pointers, are appreciated!

**Comment by jahow** [comment]
Not an isolated problem at least: https://github.com/mrdoob/three.js/issues/23733

**Comment by bunnybones1** [comment]
In theory you'll see this issue in Safari in Monterey, on M1 macbooks too, where we also found the issue to show up.

**Comment by tschaub** [comment]
It might be worth seeing if the issue goes away with `preserveDrawingBuffer = false`.  If so, we could disable this on iOS Safari 15.4.

**Comment by tschaub** [comment]
@jeroenvheel @v4lli - Can you check to see if these examples work or not:

 * https://deploy-preview-13492--ol-site.netlify.app/examples/webgl-layer-swipe.html
 * https://deploy-preview-13492--ol-site.netlify.app/examples/webgl-sea-level.html

**Comment by ahocevar** [comment]
For me, these do not work on iOS 15.4 (iPhone 13 mini)

**Comment by v4lli** [comment]
@tschaub They do not, sadly (iOS 15.4 iPhone 11 Pro, macOS 12.3 M1 MBP).

**Comment by tschaub** [comment]
Same here (just upgraded).  I’ll try another approach.

**Comment by tschaub** [comment]
I've updated #13492 to use `antialias: false` for Safari 15.4.  This makes the WebGL examples work for me on iOS 15.4 and macOS 12.3 M1.

If a fix for https://bugs.webkit.org/show_bug.cgi?id=237906 (or related issues) is not in before our next release, we may want to expand the check to include 15.5.

**Comment by jeroenvheel** [comment]
@tschaub Thanks for the fix

**Comment by v4lli** [comment]
Thanks for the fix guys, you're the best 🥰

Just a small note for everyone finding this via Google etc: the fix for #13492 is _not_ automatically applied if you're using OpenLayers inside a WkWebView since the User Agent string reported by the the WebView (`Mozilla/5.0 (iPhone; CPU iPhone OS 15_4 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Mobile/15E148`) is slightly different from Mobile Safari's (`Mozilla/5.0 (iPhone; CPU iPhone OS 15_4 like Mac OS X) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/15.4 Mobile/15E148 Safari/604.1`), and thus does not set `SAFARI_BUG_237906` to true.
